### ISSUE TITLE
GradleConnector.disconnect freeze

### ISSUE THREAD

**Issue by nskvortsov** [gradle/gradle#14634]
When a tooling client calls method `GradleConnector.disconnect()` it may take about 30s due to network issues

### Expected Behavior
Method should return fast.

### Current Behavior
Method does not return until something like default network timeout.

### Context
IntelliJ IDEA calls this method when a Gradle-based project is closed, and the effect is a full UI freeze for given time.

### Steps to Reproduce 
unfortunately, this is heavely environment specific and so far we failed to reproduce this. Relevant part of the stacktrace:
```stacktrace
	at java.base@11.0.8/sun.nio.ch.Net.poll(Native Method)
	at java.base@11.0.8/sun.nio.ch.SocketChannelImpl.pollConnected(SocketChannelImpl.java:993)
	at java.base@11.0.8/sun.nio.ch.SocketAdaptor.connect(SocketAdaptor.java:119)
	at org.gradle.internal.remote.internal.inet.TcpOutgoingConnector.tryConnect(TcpOutgoingConnector.java:81)
	at org.gradle.internal.remote.internal.inet.TcpOutgoingConnector.connect(TcpOutgoingConnector.java:54)
	at org.gradle.launcher.daemon.client.DefaultDaemonConnector.connectToDaemon(DefaultDaemonConnector.java:269)
	at org.gradle.launcher.daemon.client.DefaultDaemonConnector.maybeConnect(DefaultDaemonConnector.java:106)
	at org.gradle.launcher.daemon.client.DaemonStopClient.gracefulStop(DaemonStopClient.java:69)
	at org.gradle.tooling.internal.provider.ShutdownCoordinator.stop(ShutdownCoordinator.java:42)
	at org.gradle.tooling.internal.provider.ProviderConnection.stopWhenIdle(ProviderConnection.java:199)
	at org.gradle.tooling.internal.provider.DefaultConnection.stopWhenIdle(DefaultConnection.java:302)
	at org.gradle.tooling.internal.consumer.connection.StopWhenIdleConsumerConnection.stopWhenIdle(StopWhenIdleConsumerConnection.java:38)
	at org.gradle.tooling.internal.consumer.connection.ParameterValidatingConsumerConnection.stopWhenIdle(ParameterValidatingConsumerConnection.java:78)
	at org.gradle.tooling.internal.consumer.connection.LazyConsumerActionExecutor$1.run(LazyConsumerActionExecutor.java:112)
	at org.gradle.tooling.internal.consumer.connection.LazyConsumerActionExecutor$1.run(LazyConsumerActionExecutor.java:104)
	at org.gradle.tooling.internal.consumer.connection.LazyConsumerActionExecutor.run(LazyConsumerActionExecutor.java:130)
	at org.gradle.tooling.internal.consumer.connection.LazyConsumerActionExecutor.sendStopWhenIdleMessageToDaemons(LazyConsumerActionExecutor.java:104)
	at org.gradle.tooling.internal.consumer.connection.LazyConsumerActionExecutor.disconnect(LazyConsumerActionExecutor.java:85)
	at org.gradle.tooling.internal.consumer.connection.CancellableConsumerActionExecutor.disconnect(CancellableConsumerActionExecutor.java:50)
	at org.gradle.tooling.internal.consumer.connection.ProgressLoggingConsumerActionExecutor.disconnect(ProgressLoggingConsumerActionExecutor.java:72)
	at org.gradle.tooling.internal.consumer.connection.RethrowingErrorsConsumerActionExecutor.disconnect(RethrowingErrorsConsumerActionExecutor.java:45)
	at org.gradle.tooling.internal.consumer.async.DefaultAsyncConsumerActionExecutor.disconnect(DefaultAsyncConsumerActionExecutor.java:54)
	at org.gradle.tooling.internal.consumer.DefaultProjectConnection.disconnect(DefaultProjectConnection.java:53)
	at org.gradle.tooling.internal.consumer.DefaultGradleConnector.disconnect(DefaultGradleConnector.java:77)
	at org.jetbrains.plugins.gradle.GradleConnectorService$GradleProjectConnection.disconnect(GradleConnectorService.kt:112)
	at org.jetbrains.plugins.gradle.GradleConnectorService.disconnectGradleConnections(GradleConnectorService.kt:78)
	at org.jetbrains.plugins.gradle.GradleConnectorService.dispose(GradleConnectorService.kt:54)
...
```

### Your Environment
Tooling API 6.5.1 , IDEA 2020.3 EAP , Windows 10

**Comment by big-guy** [comment]
@nskvortsov do you know which target version of Gradle was being used?

**Comment by nskvortsov** [comment]
Target Gradle was 6.5.1

**Comment by donat** [comment]
Fixed in #15233
